scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -837,17 +836,24 @@
    :timout-hours specifies the timeout hours for lingering tasks"
   [conn driver config]
   (let [{interval-minutes :timeout-interval-minutes
-         timeout-hours :timeout-hours}
+         max-timeout-hours :max-timeout-hours
+         default-timeout-hours :default-timeout-hours
+         timeout-hours :timout-hours}
         (merge {:timeout-interval-minutes 10
                 :timeout-hours (* 2 24)}
                config)]
     (chime-at (periodic/periodic-seq (time/now) (time/minutes interval-minutes))
               (fn [now]
                 (let [db (d/db conn)
-                      lingering-tasks (get-lingering-tasks db now timeout-hours)]
+                      ;; These defaults are for backwards compatibility
+                      max-timeout-hours (or max-timeout-hours timeout-hours)
+                      default-timeout-hours (or default-timeout-hours timeout-hours)
+                      lingering-tasks (get-lingering-tasks db now max-timeout-hours default-timeout-hours)]
                   (when (seq lingering-tasks)
-                    (log/info "Starting to kill lingering jobs running more than" timeout-hours
-                              "hours. There are in total" (count lingering-tasks) "lingering tasks.")
+                    (log/info "Starting to kill lingering jobs running more than their max-runtime."
+                              {:default-timeout-hours default-timeout-hours
+                               :max-timeout-hours max-timeout-hours}
+                              "There are in total" (count lingering-tasks) "lingering tasks.")
```

Comment:
Yes, exactly that. The default is so that users don't always need to specify. The max is a bit more special case and is to prevent someone from requesting such a long time out that other factors start to cause problems. In our environment this is things like kerberos tickets expiring.